### PR TITLE
feat: likes working in backend, also added button in specific activity

### PR DESCRIPTION
**Pull request by daniellopez1999**

### CODE REVIEW

File: server/controllers/activityInteraction.ts [hunk 1]
```diff
@@ -29,3 +28,29 @@ export const saveActivityInProfile = async (req: Request, res: Response) => {
     return res.sendStatus(400).send(error);
   }
 };
+
+export const likeActivity = async (req: Request, res: Response) => {
+  try {
+    const { id, username } = req.params;
+    const user = await getUserByUserName(username);
+    const activity = await getActivitiesByID(id);
+    const userID = user?._id.toString();
```
Comment (jianingroja) [inline server/controllers/activityInteraction.ts:37]: user.id will return the string of user._id

File: server/router.ts [hunk 2]
```diff
@@ -71,5 +74,6 @@ router.post('/generator', generateActivity);
 router.get('/generator', generateActivity);
 
 router.post('/savepost-in-user/:username/:id', saveActivityInProfile);
+router.post('/saveLike/:username/:id', likeActivity);
```
Comment (jianingroja) [inline server/router.ts:77]: /save-like/:username/:id

File: client/src/services/activity.ts [hunk 3]
```diff
@@ -71,3 +71,22 @@ export const saveActivityInProfile = async (
     console.error(error);
   }
 };
+
+export const saveLike = async (username: string, activityID: string) => {
+  try {
+    const response = await fetch(
+      `${BASE_URL}/saveLike/${username}/${activityID}`,
```
Comment (jianingroja) [inline client/src/services/activity.ts:78]: save-like

File: client/src/components/Activity/SpecificActivity.tsx [hunk 4]
```diff
@@ -47,7 +57,9 @@ const SpecificActivity = () => {
       <div className="status">
         <p>{activityData?.activityInfo.likes.length} likes</p>
         <div className="actions">
-          <p>like</p>
+          <p>
+            <button onClick={() => like()}>like</button>
```
Comment (jianingroja) [inline client/src/components/Activity/SpecificActivity.tsx:61]: how about unlike? we can also discuss if it is necessary to implement unlike

**Review by ZarinaPoyet — APPROVED** [review]
lgtm👍

**Pull request merged**